### PR TITLE
coherence with principles stated in #63

### PR DESCRIPTION
**Pull request by JohnRDOrazio**
Just to keep all `Trans` components nice and tidy. I just noticed that the `Trans` components in `Prayers.js` have nested elements: this won't work when automatically scanning such `Trans` components. So simply for coherence with our non-nested elements principle, so as to avoid indexed nodes being generated on a scan, this nests the `Trans` components in `<p>`s rather than the `<p>`s within the `Trans` components.

### CODE REVIEW

**Comment by JohnRDOrazio** [comment]
I hope you find commit 4053d02 acceptable, I figure it's useless to translate the exact same text twice in two different contexts. I think it should be acceptable to recycle the translation from `Prayers.js`...

File: src/Walkthrough.js [hunk 1]
```diff
@@ -22,23 +22,7 @@ class WalkthroughComponent extends React.Component {
...
-              I firmly intend with Your help to do penance, to sin no more, and
-              to avoid whatever leads me to sin.
-            </p>
-            <p>
-              Jesus Christ suffered and died for us. In His name, my God, have
-              mercy.
-            </p>
-          </Trans>
-        </UserBubble>
+        <UserBubble>{t("prayers.act_of_contrition")}</UserBubble>
```
Comment (kas-catholic) [inline src/Walkthrough.js:25]: > I hope you find commit https://github.com/kas-catholic/confessit-web/commit/4053d02c43eb646733288f6074a5107a38aac197 acceptable, I figure it's useless to translate the exact same text twice in two different contexts.

Seems fine I guess, I don't have strong feelings either way.

----

But this is the wrong translation, it should be `act_of_contrition_text` to get the complete prayer, not just the title.

image


https://deploy-preview-67--confessit-web.netlify.app/

**Review by kas-catholic — CHANGES_REQUESTED** [review]
Requesting changes because of the incorrect translation key for the act of contrition.

----

I'm fine merging this after that's fixed, but I don't understand why we need to. In #63, you say

> Any nodes within Trans components that have attributes, or are nested, or do not have just text nodes as children, will be converted to indexed nodes

I believe you got that from [these docs](https://react.i18next.com/latest/trans-component#usage-with-simple-html-elements-like-less-than-br-greater-than-and-others-v10.4.0). But my reading of the linked docs is that a component like this `<Trans><p>foo</p><p>bar</p></Trans>` _is not nested_ according to those same linked docs. Indeed, `<p>` is listed as an example under "elements that will be readable in translation strings", and it _was_ readable as `<p>` in the existing translations. It is included in the default setting of `transKeepBasicHtmlNodesFor`.

Am I missing something about why its important that we change all these `<p>` tags to `<br />`? `<p>` is preferable from a css/styling perspective...

**Comment by kas-catholic** [comment]
Re-reading your description, I think I get it now. The important thing is to change `<Trans><p>...</p></Trans>` to `<p><Trans></Trans></p>`. And the problem with the nested tags is having `<p>...<br />...</p>` inside a translation. That makes sense to me, let's do that!

**Comment by JohnRDOrazio** [comment]
> Requesting changes because of the incorrect translation key for the act of contrition.
> 
> I'm fine merging this after that's fixed, but I don't understand why we need to. In #63, you say
> 
> > Any nodes within Trans components that have attributes, or are nested, or do not have just text nodes as children, will be converted to indexed nodes
> 
> I believe you got that from [these docs](https://react.i18next.com/latest/trans-component#usage-with-simple-html-elements-like-less-than-br-greater-than-and-others-v10.4.0). But my reading of the linked docs is that a component like this `<Trans><p>foo</p><p>bar</p></Trans>` _is not nested_ according to those same linked docs. Indeed, `<p>` is listed as an example under "elements that will be readable in translation strings", and it _was_ readable as `<p>` in the existing translations. It is included in the default setting of `transKeepBasicHtmlNodesFor`.
> 
> Am I missing something about why its important that we change all these `<p>` tags to `<br />`? `<p>` is preferable from a css/styling perspective...

Only change there was in the Italian: I had originally used the most common version of the act of contrition on `Walkthrough.js`, and two other versions in `Prayers.js`. But since I removed the `Trans` component with the act of contrition from `Walkthrough.js`, I figured it would probably be a good idea to keep the most common version in any case, so I substituted the first one in `Prayers.js` with the most common one, and the second alternate version with the second most common version offered by the Ritual in Italian. Third alternate version in Italian wound up getting ditched in this process.

**Comment by JohnRDOrazio** [comment]
And in fact the whole reason I even noticed any of this was when doing another run with `i18n-scanner`, and seeing that it still output indexed tags for certain `Trans` components

File: src/Walkthrough.js (hunk 1 above)
Comment (JohnRDOrazio) [reply src/Walkthrough.js]: dang didn't even double check that! I just switched to `act_of_contrition_text`, to find another problem: when using `Trans` components, HTML is preserved. But when using a `t()` function to retrieve a translation, html is not preserved. Only way to keep it, is to use a wrapper `<div>` setting the `dangerouslySetInnerHtml` property with the text of the translation and the indication to not escape html (`interpolation: { escapeValue: false }`).

Does this still work, or is it starting to get messy?

Comment (JohnRDOrazio) [reply src/Walkthrough.js]: Well I guess there is no such thing as an empty self-closing `Trans` component, it was causing tests to fail. I had seen it as an example on [stackoverflow](https://stackoverflow.com/a/72177021/394921), but I guess it isn't quite correct. So I'm just using an exact copy of the `Trans` component from `Prayers.js`, this seems to work fine and tests pass...

**Review by kas-catholic — APPROVED** [review]

**Pull request merged**